modules/AOW_WorkFlow/aow_utils.php

```diff
@@ -305,14 +305,53 @@ function getValidFieldsTypes($module, $field){
 }
 
 
-function getModuleField($module, $fieldname, $aow_field, $view='EditView',$value = '', $alt_type = '', $currency_id = '', $params= array()){
-    global $current_language, $app_strings, $app_list_strings, $current_user, $beanFiles, $beanList;
+function getModuleField(
+    $module,
+    $fieldname,
+    $aow_field,
+    $view='EditView',
+    $value = '',
+    $alt_type = '',
+    $currency_id = '',
+    $params= array()
+){
+    global $current_language;
+    global $app_strings;
+    global $app_list_strings;
+    global $current_user;
+    global $beanFiles;
+    global $beanList;
 
     // use the mod_strings for this module
     $mod_strings = return_module_language($current_language,$module);
 
+    // if aor condition
+    if(strstr($aow_field, 'aor_conditions_value') !== false) {
+        // get aor condition row
+        $aor_row = str_replace('aor_conditions_value', '', $aow_field);
+        $aor_row = str_replace('[', '', $aor_row);
+        $aor_row = str_replace(']', '', $aor_row);
+        // set the filename for this control
+        $file = create_cache_directory('modules/AOW_WorkFlow/')
+            . $module
+            . $view
+            . $alt_type
+            . $fieldname
+            . $aor_row
+            . '.tpl';
+    } else {
+        //  its probably result of the report
+        // set the filename for this control
+        $file = create_cache_directory('modules/AOW_WorkFlow/')
+            . $module
+            . $view
+            . $alt_type
+            . $fieldname
+            . '.tpl';
+    }
+
     // set the filename for this control
-    $file = create_cache_directory('modules/AOW_WorkFlow/') . $module . $view . $alt_type . $fieldname . '.tpl';
+    $file = create_cache_directory('modules/AOW_WorkFlow/') . $module . $view . $alt_type . $fieldname . $aor_row . '.tpl';
```

Comment:
A new tpl for each row?

Reply:
Yes, this ensures that the id and name are correct.